### PR TITLE
Fix Robin condition for ESI and Foundation versions

### PR DESCRIPTION
**Pull request by philipcardiff**
Merge @ztukovic fixes for Robin condition when used with ESI (OpenFOAM.com) and Foundation (OpenFOAM.org) versions.

This also related to
- https://github.com/solids4foam/solids4foam/issues/26
- https://github.com/solids4foam/solids4foam/pull/29

### CODE REVIEW

**Comment by philipcardiff** [comment]
May be of interest to [SimoMor87](https://github.com/SimoMor87) and @iagolessa

File: tutorials/solids/thermoelasticity/hotSphere/README.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 ---
 sort: 1
+output:
+  html_document:
+    df_print: paged
```
Comment (philipcardiff) [inline tutorials/solids/thermoelasticity/hotSphere/README.md:5]: @ztukovic : I guess this option changes how the README is displayed. Do you know what this is needed for?

Comment (ztukovic) [reply tutorials/solids/thermoelasticity/hotSphere/README.md:5]: Sorry I have opened this file with RStudio and made these modification unintentionally. You can restore old version

Comment (philipcardiff) [reply tutorials/solids/thermoelasticity/hotSphere/README.md:5]: Ah, OK. I will revert it.

... maybe this helps in some way: I will check this some other day.

**Comment by philipcardiff** [comment]
@ztukovic: to check everything is as expected, I ran the 3dTube case in the following six ways (all with pimple):
- FE41: 3dTube and 3dTubeRobin
- OFv2012: 3dTube and 3dTubeRobin
- OF9: 3dTube and 3dTubeRobin

The Robin case is much faster for all versions (as expected), although the convergence is different between the versions; here are the number of iterations plots:
[fe41_robin.pdf](https://github.com/solids4foam/solids4foam/files/12343274/fe41_robin.pdf)
[v2012_robin.pdf](https://github.com/solids4foam/solids4foam/files/12343275/v2012_robin.pdf)
[of9_robin.pdf](https://github.com/solids4foam/solids4foam/files/12343277/of9_robin.pdf)

Here are the wall clock times:
- FE41 3dTube: 19489 s
- FE41 3dTubeRobin: 2750 s
- OFv2012 3dTube: 32327 s
- OFv2012 3dTubeRobin: 3698 s
- OF9: 3dTube: 32252 s
- OF9 3dTubeRobin: 9628 s

Before I merge these, are these results as you expect? It looks fine to me: the Robin condition is accelerating the convergence as expected, although it is interesting that there is such a difference between the versions.

By the way, I might merge the 3dTube and 3dTubeRobin cases and use symbolic links for easy switching between the options.

**Comment by ztukovic** [comment]
@philipcardiff, differences are probabli related to backward scheme. I think you will get similar rezultat with Euler scheme. I will try to fix this problem in following days

**Comment by philipcardiff** [comment]
> @philipcardiff, differences are probabli related to backward scheme. I think you will get similar rezultat with Euler scheme. I will try to fix this problem in following days

No problem; I will merge the changes as they are, as Robin still works better in all three versions. If there are future fixes, they can be added when they are ready.

**Comment by philipcardiff** [comment]
For future reference: https://github.com/solids4foam/solids4foam/pull/60#issuecomment-1682094626

**Comment by philipcardiff** [comment]
@ztukovic : this branch is now merged, so you can continue to work from the development branch.

**Pull request merged**